src/DotNetWorker.Grpc/FunctionMetadata/FunctionMetadataJsonReader.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Threading.Tasks;
+using System.Text.Json;
+using Microsoft.Azure.Functions.Worker.Grpc.Messages;
+
+namespace Microsoft.Azure.Functions.Worker
+{ 
+    internal class FunctionMetadataJsonReader
+    {
+        private readonly string _directory;
+        private const string FileName = "functions.metadata";
+
+        public FunctionMetadataJsonReader(string directory)
+        {
+            _directory = directory ?? throw new ArgumentNullException(nameof(directory));
+        }
+
+        public virtual async Task<ImmutableArray<FunctionLoadRequest>> ReadMetadataAsync()
+        {
+            string metadataFile = Path.Combine(_directory, FileName);
+
+            if (File.Exists(metadataFile))
+            {
+                using (var fs = File.OpenRead(metadataFile))
+                {
+                    var jsonMetadataList = await JsonSerializer.DeserializeAsync<JsonElement>(fs);
+
+                    var functionRequests = new List<FunctionLoadRequest>(jsonMetadataList.GetArrayLength());
+
+                    foreach (var jsonMetadata in jsonMetadataList.EnumerateArray())
+                    {
+                        var options = new JsonSerializerOptions();
```

Comment:
Could we use a single instance of `JsonSerializerOptions` and use that? No need to create a new instance for each iteration of the loop.